src/bootstrap/bootstrap.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2015-2016 The Rust Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution and at
+# http://rust-lang.org/COPYRIGHT.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import argparse
+import contextlib
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+
+def get(url, path, quiet=False):
+    if quiet:
+        print("downloading " + url)
+    # see http://serverfault.com/questions/301128/how-to-download
+    if sys.platform == 'win32':
+        run(["PowerShell.exe", "/nologo", "-Command",
+             "(New-Object System.Net.WebClient).DownloadFile('" + url +
+                "', '" + path + "')"], quiet=quiet)
+    else:
+        run(["curl", "-o", path, url], quiet=quiet)
+
+def unpack(tarball, dst, quiet=False, filter=None):
```

Comment:
Not a huge deal, but FYI, `filter` here is shadowing the global `filter` function